app/models/location.rb

```diff
@@ -1,5 +1,37 @@
+require 'geokit'
+
 class Location < ActiveRecord::Base
   belongs_to :user
 
   validates :user, presence: true
+
+  after_save :reindex_dishes
+
+  def reindex_dishes
+    user.dishes.each { |dish| dish.reindex }
+  end
+
+  # only update latlng when moved distance > 0.5 mile
+  # this is to reduce the cost whenever location is changed, have to reindex dishes (ElasticSearch)
+  def update_latlng(new_lat, new_lng)
+    new_lat = new_lat.try(:to_f) if !new_lat.blank?
```

Comment:
Can lines 17 and 18 be simplified? I'm not exactly sure I've fully understood what's needed here though.